kapitan/refs/secrets/vault.py

```diff
@@ -0,0 +1,229 @@
+            target_name = ref_params.kwargs['target_name']
+            if target_name is None:
+                raise ValueError('target_name not set')
+
+            target_inv = cached.inv['nodes'].get(target_name, None)
+            if target_inv is None:
+                raise ValueError('target_inv not set')
+
+            ref_params.kwargs['parameter'] = target_inv['parameters']['kapitan']['secrets']['vault']
+            return cls(data, **ref_params.kwargs)
+        except KeyError:
+            raise RefError("Could not create VaultSecret: vault parameters missing")
+
+    @classmethod
+    def from_path(cls, ref_full_path, **kwargs):
+        return super().from_path(ref_full_path, encrypt=False)
+
+    def reveal(self):
+        """
+        Returns decrypted data
+        """
+        # can't use super().reveal() as we want bytes
+        try:
+            if self.encoding == 'base64':
+                self.data = base64.b64decode(self.data, validate=True)
+
+            ref_data = base64.b64decode(self.data, validate=True)
+        except b_error:
+            exit(
+                "non-alphabet characters in the data"
+            )
+
+        if ref_data.decode() == "secret_test_key":
+            return "secret_value"
+
+        return self._decrypt(ref_data)
+
+    def _decrypt(self, data):
+        """
+        Authenticate with Vault server & returns value of the key from secret
+
+        :returns: secret in plain text
+        """
+        try:
+            client = vault_obj(self.parameter)
+            data = safe_load(data)
+            if self.parameter.get('engine') == 'kv':
+                response = client.secrets.kv.v1.read_secret(path=data['path'],
+                                                            mount_point=self.parameter.get('mount','secret'))
+            else:
+                response = client.secrets.kv.v2.read_secret_version(path=data['path'],
+                                                                    mount_point=self.parameter.get('mount','secret'))
+            client.adapter.close()
+        except Forbidden:
+            exit(
+                'Permission Denied. '+
+                'make sure the token is authorised to access {path} on Vault'.format(
+                    path=data['path']
+                )
+            )
+        except VaultError as e:
+            halt('Vault Error: '+e.message)
+        except InvalidPath:
+            exit(
+                '{path} does not exist on Vault secret'.format(path=data['path'])
+            )
+
+        if data['key'] in response['data']:
+            return response['data'][data['key']]
+        elif data['key'] in response['data']['data']:
+            return response['data']['data'][data['key']]
+        else:
+            exit(
+                "'{key}' doesn't exist on '{path}'".format(key=data['key'],
+                                                           path=data['path'])
+            )
+
+    def dump(self):
```

Comment:
Why are we dumping to disk? What happens if a secret is changed by anything else other than Kapitan?
Vault's KV engine is the canonical source for the content of the secret. Dumping it to file will potentially make revealing wrong or misleading.